dem_handler/utils/spatial.py

```diff
@@ -509,3 +514,61 @@ def crop_datasets_to_bounds(
             # )
 
     return dem_array, dem_profile
+
+
+def check_dem_type_in_bounds(dem_type: str, bounds: BBox) -> bool:
+    """Check if the specified dem has data within the provided bounds. The provided dem_type is matched to either the
+    Copernicus Global 30m DEM or REMA DEM (currently implemented options). True is returned if the provided bounds
+    intersect with any tiles of the specified DEM.
+
+    Parameters
+    ----------
+    dem_type : str
+        dem type to check. Can be variations of REMA and COP_30. e.g. REMA_32, REMA_10,
+        cop30m, cop_30m, cop_glo30.
+    bounds : BBox
+        Bounds to check if data exists
+
+    Returns
+    -------
+    bool
+        True if the bounds intersects a tile, False otherwise.
+
+    Raises
+    -------
+    ValueError
+        If the provided dem_type cannot be matched to either the Copernicus 30m global DEM or the REMA dem.
+    """
+
+    if type(bounds) == BoundingBox:
+        bounds = bounds.bounds
+
+    dem_type_match = dem_type.upper()
+    if "COP" and "30" in dem_type_match:
+        dem_index_path = COP30_GPKG_PATH
+        dem_type_formal = "Copernicus 30m global DEM"
+    elif "REMA" in dem_type_match:
+        dem_index_path = REMA_GPKG_PATH
+        dem_type_formal = "REMA DEM"
+    else:
+        raise ValueError(
+            f"DEM type `{dem_type}` could not be matched to either the Copernicus 30m global DEM or the REMA DEM"
+        )
+
+    logger.info(f"Checking if bounds intersect with tiles of the {dem_type_formal}")
+    logger.info(f"Searching {COP30_GPKG_PATH}")
+
+    gdf = gpd.read_file(dem_index_path)
+    bounding_box = shapely.geometry.box(*bounds)
+
+    if gdf.crs is not None:
```

Comment:
Do you want to do something if the gdf.crs is None? Throw an error or a warning?

Reply:
Apologies, I missed that. Perhaps add a comment to make it clear?